pyopae/pyhandle.cpp

```diff
@@ -98,7 +99,10 @@ const char *handle_doc_context_enter() {
   )opaedoc";
 }
 
-handle::ptr_t handle_context_enter(handle::ptr_t hnd) { return hnd; }
+handle::ptr_t handle_context_enter(handle::ptr_t hnd) {
+  buffer_registry::instance().register_handle(hnd);
```

Comment:
Where is the handle from?

Reply:
Yes, handle here is the C++ handle class in the C++ API which is wrapped by the Python bindings (as defined in opae.cpp) using pybind11.
The opae.cpp file also defines \_\_enter\_\_ and \_\_exit\_\_ methods for the handle class which get called when entering a context in Python.
```Python
with opae.fpga.open(tok, flags) as h: # this calls the __enter__ function
    # do stuff with h while it is open
    # you can allocate buffers with the handle here
    buf = opae.fpga.allocate_shared_buffer(h, size)
# __exit__ is called when the context is closed (when this scope exits)
assert not h # This should assert should pass as the 
assert buf.size() == 0 # the buffer should be released by the context manager __exit__